shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -30,54 +34,98 @@ pub struct CassandraPeersRewrite {
 impl Transform for CassandraPeersRewrite {
     async fn transform<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
         // Find the indices of queries to system.peers & system.peers_v2
-        let system_peers = message_wrapper
+        // we need to know which columns in which CQL queries in which messages have system peers
+        let mut column_names: HashMap<usize, Vec<String>> = HashMap::new();
+
+        message_wrapper
             .messages
             .iter_mut()
             .enumerate()
-            .filter_map(|(i, m)| if is_system_peers(m) { Some(i) } else { None })
-            .collect::<Vec<_>>();
+            .filter_map(|(i, m)| {
+                let sys_peers = extract_native_port_column(m);
+                if sys_peers.is_empty() {
+                    None
+                } else {
+                    Some((i, sys_peers))
+                }
+            })
+            .for_each(|(k, mut v)| {
+                if let Some(x) = column_names.get_mut(&k) {
+                    x.append(&mut v);
+                } else {
+                    column_names.insert(k, v);
+                }
+            });
 
         let mut response = message_wrapper.call_next_transform().await?;
 
-        for i in system_peers {
-            rewrite_port(&mut response[i], self.port);
+        for (idx, name_list) in column_names {
+            rewrite_port(&mut response[idx], &name_list, self.port);
         }
 
         Ok(response)
     }
 }
 
-fn is_system_peers(message: &mut Message) -> bool {
-    if let Some(Frame::Cassandra(_)) = message.frame() {
-        if let Some(namespace) = message.namespace() {
-            if namespace.len() > 1 {
-                return namespace[0] == "system" && namespace[1] == "peers_v2";
+/// determine if the message contains a SELECT from `system.peers_v2` that includes the `native_port` column
+/// return a list of (statement index, column index) pairs
+fn extract_native_port_column(message: &mut Message) -> Vec<String> {
+    let mut result: Vec<String> = vec![];
+    if let Some(Frame::Cassandra(cassandra)) = message.frame() {
+        if let CassandraOperation::Query { query, .. } = &cassandra.operation {
+            for cql_statement in &query.statements {
+                let statement = &cql_statement.statement;
+                if let CassandraStatement::Select(select) = &statement {
+                    if let Some(table_name) = CQLStatement::get_table_name(statement) {
+                        if table_name.to_string().eq("system.peers_v2") {
```

Comment:
We can just use `select.table_name` here instead of having to call get_table_name.

Reply:
If we write this `if` like:
```rust
                    if select
                        .table_name
                        .keyspace
                        .as_ref()
                        .map(|k| k == "system")
                        .unwrap_or(false)
                        || select.table_name.name == "peers_v2"
```
then we can avoid allocating the string.
Maybe a helper on FQName could let us do so in a cleaner way.

Reply:
we can use == instead of eq

Reply:
I dont see how that happens here?
Is it a proposed thing that hasnt happened yet?

Reply:
oh yeah, that should work!

Reply:
Actually I dont think that would work.
I think we cant create an arbitrary const String in rust yet.
Might be wrong though.

Reply:
I did the delete the comment shortly after posting it because I decided it wasnt high priority.
Up to you if you want to investigate it further.